Objective-C/Tests/CollectionTest.m

```diff
@@ -648,6 +648,23 @@ - (void) testCollectionChangeListener {
     AssertEqual(changeListenerFired, 0);
 }
 
+- (void) testCollectionChangeListenerWithoutRemoveToken {
```

Comment:
Seems like it could use comments since the actual assertions that are important are handled in the teardown (same for all new tests in this PR)?